### PR TITLE
fix: use the two-steps log in on hosted

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: None
Ticket: MEN-6823

### CODE REVIEW

File: src/js/components/login/login.test.js [hunk 1]
```diff
@@ -47,6 +47,13 @@ describe('Login Component', () => {
     const { rerender } = render(ui, { preloadedState });
     await act(async () => {
       await user.type(screen.getByLabelText(/your email/i), 'something-2fa@example.com');
+    });
+    await act(async () => {
+      await user.click(screen.getByRole('button', { name: /Log in/i }));
+    });
```
Comment (mzedel) [inline src/js/components/login/login.test.js:53]:
```suggestion
    await act(async () => {
      await user.type(screen.getByLabelText(/your email/i), 'something-2fa@example.com');
      await user.click(screen.getByRole('button', { name: /Log in/i }));
    });
```

the call to `act` can take a bit of time to tear down and cause weird problems in test execution, thus the RTL recommendation is to bundle them similar to an active phase of a user interacting with a page

**Review by mzedel — APPROVED** [review]

File: src/js/components/login/login.test.js (hunk 1 above)
Comment (tranchitella) [reply src/js/components/login/login.test.js:53]: This code fails with:

`
  ● Login Component › works as intended

    Unable to perform pointer interaction as the element has `pointer-events: none`:

    BUTTON(label=Log in)

      48 |     await act(async () => {
      49 |       await user.type(screen.getByLabelText(/your email/i), 'something-2fa@example.com');
    > 50 |       await user.click(screen.getByRole('button', { name: /Log in/i }));
         |       ^
      51 |     });
      52 |     expect(loginSpy).toHaveBeenCalled();
      53 |     await waitFor(() => rerender(ui));
`

Comment (tranchitella) [reply src/js/components/login/login.test.js:53]: the pipeline is passing and you approved the PR, @mzedel.
If I need to rework this particular section, let me know and I will follow up with a dedicated PR.

Comment (mzedel) [reply src/js/components/login/login.test.js:53]: Thanks for the explanation! The recommendation would be to be explicit and do a
`await waitFor(() =>  expect(screen.getByRole('button', { name: /log in/i })).not.toBeDisabled());`
or often an
`await waitFor(() => rerender(ui));`
also works...

but OTOH - if it works I won't mind... and absolutely: once approved always merge ahead!

**Pull request merged**